R/normalize_nmr.R

```diff
@@ -0,0 +1,393 @@
+            edata_new <- edata_new * backtransform_value
+            # re-log2 to the data
+            # edata_new < log2(edata_new) # this doesn't work...spits out matrix of FALSE and the edata_new itself doesn't get changed from before
+            edata_new <- apply(edata_new, 2, log2)
+          }else{
+            if(data_scale == "log10"){
+              # un-log10 the data and divide by sample property
+              edata_new <- 10^edata_new / matrix(sample_property, nrow = nrow(edata_new), ncol = ncol(edata_new), byrow = TRUE)
+              # do the backtransform
+              edata_new <- edata_new * log10(backtransform_value)
+              # re-log10 the data
+              # edata_new <- log10(edata_new)
+              edata_new <- apply(edata_new, 2, log10)
+            }else{
+              if(data_scale == "ln"){
+                # un ln the data and divide by sample property
+                edata_new <- exp(edata_new) / matrix(sample_property, nrow = nrow(edata_new), ncol = ncol(edata_new), byrow = TRUE)
+                # do the backtransform
+                edata_new <- edata_new * ln(backtransform_value)
+                # re-ln the data
+                # edata_new <- log(edata_new)
+                edata_new <- apply(edata_new, 2, log)
+              }else{
+                stop("data_scale is not recognized as valid. Check attributes(omcisData)$data_info$data_scale")
+              }
+            }
+          }
+        }else{ # backtransform == FALSE
+          if(data_scale == "log2"){
+            # un-log2 the data and divide by sample property
+            edata_new <- 2^edata_new / matrix(sample_property, nrow = nrow(edata_new), ncol = ncol(edata_new), byrow = TRUE)
+            # re-log2 to the data
+            # edata_new < log2(edata_new)
+            edata_new <- apply(edata_new, 2, log2)
+          }else{
+            if(data_scale == "log10"){
+              # un-log10 the data and divide by sample property
+              edata_new <- 10^edata_new / matrix(sample_property, nrow = nrow(edata_new), ncol = ncol(edata_new), byrow = TRUE)
+              # re-log10 the data
+              # edata_new <- log10(edata_new)
+              edata_new <- apply(edata_new, 2, log10)
+            }else{
+              if(data_scale == "ln"){
+                # un ln the data and divide by sample property
+                edata_new <- exp(edata_new) / matrix(sample_property, nrow = nrow(edata_new), ncol = ncol(edata_new), byrow = TRUE)
+                # re-ln the data
+                # edata_new <- log(edata_new)
+                edata_new <- apply(edata_new, 2, log)
+              }else{
+                stop("data_scale is not recognized as valid. Check attributes(omcisData)$data_info$data_scale")
+              }
+            }
+          }
+        }
+        
+        edata_new <- data.frame(edata_new) # convert back to data.frame since the apply statements above switch it to matrix, array
```

Comment:
I think we'll need to pass a check.names argument here?

Reply:
Done